pkg/apis/sessions/session_state.go

```diff
@@ -16,8 +16,9 @@ import (
 
 // SessionState is used to store information about the currently authenticated user session
 type SessionState struct {
-	CreatedAt *time.Time `msgpack:"ca,omitempty"`
-	ExpiresOn *time.Time `msgpack:"eo,omitempty"`
+	Authenticated bool       `msgpack:"au,omitempty"`
```

Comment:
`a` isn't taken already and this is 1-word, so you can claim that for efficiency.

Someday maybe we should convert this to protobuf 😅  -- Msgpack was at least a quick win cutting the cookie size in half from the old JSON encoding.

Reply:
I didn't fully follow in the PR, is this only being used during the internal flow as you are checking the authentication status in the sign in? If so, this can likely be not-serialized similar to Clock below.

If this is checked as sessions are passed later on post signin, then carry on with it serialized into the cookie.

Reply:
It is checked in all sessions later to determine if we should proxy.

Reply:
Is that needed? Is there a scenario you intend to mint a session if the PKSE code challenge failed?

Otherwise, I would think if PKSE is used and fails, instead of saying `Authenticated = false`, we don't create a session and error during that flow in `oauthproxy.go`?

This isn't that big of a deal as far as session struct size bloat (tokens are our biggies). We've just had so many issues with the session struct & cookie size in the past that I prefer to keep it lean & simple and not overload it with logic.

Reply:
Resolving this conversation as it has continued on another comment below